modules/motions/ui/MotionFormStartNew/Parts/StartNewNodeOperatorLimitIncrease.tsx

```diff
@@ -52,19 +52,19 @@ export const formParts = ({
       const nodeOperators = useNodeOperatorsList(registryType)
       const keysInfo = useNodeOperatorKeysInfo(registryType)
 
-      const [operatorId, currentNodeOperator] = useMemo(() => {
-        if (!walletAddress) return [undefined, null]
-        const idx = nodeOperators.data?.list.findIndex(
+      const currentNodeOperator = useMemo(() => {
+        if (!walletAddress || !nodeOperators.data) {
+          return null
+        }
+        return nodeOperators.data.find(
           o =>
             utils.getAddress(o.rewardAddress) ===
             utils.getAddress(walletAddress),
         )
-        const operator = idx !== undefined && nodeOperators.data?.list[idx]
-        return [idx, operator]
-      }, [walletAddress, nodeOperators])
+      }, [nodeOperators.data, walletAddress])
 
       const isNodeOperatorConnected =
-        !nodeOperators.data?.isRegistrySupported || Boolean(currentNodeOperator)
+        currentNodeOperator && walletAddress && nodeOperators.data
```

Comment:
it looks like `Boolean(currentNodeOperator)`, bc if `walletAddress && nodeOperators.data` is false  than `currentNodeOperator` will be null